src/content/1.7/development/components/grid-component.md

```diff
@@ -7,229 +7,404 @@ weight: 2
+you can implement your own Grid Data factory.
+
+We will be using `DoctrineGridDataFactory` to create data for our Grid. When using `DoctrineGridDataFactory` you have
+to implement `DoctrineQueryBuilderInterface` which will be used by data factory to build Doctrine queries.
+
+{{% notice note %}}
+When implementing `DoctrineQueryBuilderInterface` it is recommended to use `PrestaShop\PrestaShop\Core\Grid\Query\AbstractDoctrineQueryBuilder`
+as it provides access to Doctrine `Connection` and database tables prefix.
+{{% /notice %}}
+
+```php
+use PrestaShop\PrestaShop\Core\Grid\Query\AbstractDoctrineQueryBuilder;
+
+final class ProductQueryBuilder extends AbstractDoctrineQueryBuilder
 {
     /**
-     * Get query that searches grid rows
-     *
-     * @param SearchCriteriaInterface|null $searchCriteria
-     *
-     * @return QueryBuilder
+     * @var int
+     */
+    private $contextLangId;
+
+    /**
+     * @var int
      */
-    public function getSearchQueryBuilder(SearchCriteriaInterface $searchCriteria = null);
+    private $contextShopId;
+
     /**
-     * Get query that counts grid rows
-     *
-     * @param SearchCriteriaInterface|null $searchCriteria
-     *
-     * @return QueryBuilder
+     * @param Connection $connection
+     * @param string $dbPrefix
+     * @param int $contextLangId
+     * @param int $contextShopId
      */
-    public function getCountQueryBuilder(SearchCriteriaInterface $searchCriteria = null);
+    public function __construct(Connection $connection, $dbPrefix, $contextLangId, $contextShopId)
+    {
+        parent::__construct($connection, $dbPrefix);
 
-```
+        $this->contextLangId = $contextLangId;
+        $this->contextShopId = $contextShopId;
+    }
 
-Once the Query Builder is defined, we have all we need to create the grid.
+    // Get Search query builder returns QueryBuilder that is used to fetch filtered, sorted and paginated data from database.
+    // This query builder is also used to get SQL query that was executed.
+    public function getSearchQueryBuilder(SearchCriteriaInterface $searchCriteria)
+    {
+        $qb = $this->getBaseQuery();
+        $qb->select('p.id_product, p.reference, pl.name')
+            ->orderBy(
+                $searchCriteria->getOrderBy(),
+                $searchCriteria->getOrderWay()
+            )
+            ->setFirstResult($searchCriteria->getOffset())
+            ->setMaxResults($searchCriteria->getLimit());
+    
+        foreach ($searchCriteria->getFilters() as $filterName => $filterValue) {
+            if ('id_product' === $filterName) {
+                $qb->andWhere("p.id_product = :$filterName");
+                $qb->setParameter($filterName, $filterValue);
+
+                continue;
+            }
+
+            $qb->andWhere("$filterName LIKE :$filterName");
+            $qb->setParameter($filterName, '%'.$filterValue.'%');
+        }
+
+        return $qb;
+    }
+    
```

Comment:
`getCountQueryBuilder` should integrate the filters clauses, the aim is not to have the total count but the contextual count (with filters)
otherwise how do you paginate when you use a filter? only the `setMaxResults` `setOffset` part should be restricted to `getSearchQueryBuilder`

Reply:
I guess you are right, it should integrate filters. :+1: